idx/memory/tag_query_id_filter.go

```diff
@@ -88,9 +87,7 @@ func newIdFilter(expressions tagquery.Expressions, ctx *TagQueryContext) *idFilt
 		// in a metric that gets filtered, compared to doing a full tag index lookup
 		// to check whether a metric has one of the necessary meta tags associated
 		// with it.
-		optimizeForOnlyEqualOperators := !invertSetOfMetaRecords
 		var metaRecordFilters []tagquery.MetricDefinitionFilter
```

Comment:
> Is that comment referring to not populating metaRecordFilters?

yes, if all meta records which are used to filter by a meta tag only use `=` expressions, then we can skip generating the "normal" `metaRecordFilters` which do lookups on the tag index to decide whether a metric may pass the filter and instead of the `metaRecordFilters` we generate a filter function by calling `metaRecordFilterBySetOfValidValues()`/`metaRecordFilterBySetOfValidValueSets()` which uses a fixed set of tag/value combinations to compare the metrics being filtered against. This is faster but it only works if the expressions underlying meta records use `=` operations.

Reply:
I may extend this optimization so it is also used for `!=` operations on meta tags where all underlying meta records only use `=` for the issue https://github.com/grafana/metrictank/issues/1871
Currently this optimizations is only used if the meta tag filter condition uses a "positive" operator such as `=` or `=~`, but not `!=` or `!~`